core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1342,31 +1357,25 @@ class KafkaApis(val requestChannel: RequestChannel,
         )
       )
     } else {
-      val encodedGroupInstanceId = joinGroupRequest.data().groupInstanceId
-      val groupInstanceId =
-        if (encodedGroupInstanceId == null ||
-        config.interBrokerProtocolVersion < KAFKA_2_3_IV0)
-          None
-        else
-          Some(encodedGroupInstanceId)
+      val groupInstanceId = getGroupInstanceId(joinGroupRequest.data.groupInstanceId)
```

Comment:
The logic here seems a bit duplicated: in `getGroupInstanceId` our logic is to strip off the instance id, and above we actually reject it directly.

I think we are going for the second option here, right? If yes we should just remove the logic inside `getGroupInstanceId`.

Reply:
Yep, thanks for the catch!